### PR TITLE
[Enhancement] One script to export bert for classification/regression/QA

### PR DESCRIPTION
**Pull request by eric-haibin-lin**
## Description ##
This PR unifies `static_export_squad.py` and `static_export_base.py` by introducing a "task" argument to `export.py`. Currently it supports: classification, regression and question_answering. 

The PR also renames the StaticXXX classes to HybridXXX classes, in case we change the implementation with control flow operators to support variable length inputs in the future.

### Essentials ###
- [x] PR's title starts with a category (e.g. [BUGFIX], [MODEL], [TUTORIAL], [FEATURE], [DOC], etc)
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage
- [x] Code is well-documented

### Changes ###
- [ ] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

**Comment by mli** [comment]
Job PR-705/1 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-705/1/index.html

**Comment by mli** [comment]
Job PR-705/2 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-705/2/index.html

**Comment by mli** [comment]
Job PR-705/3 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-705/3/index.html

**Comment by mli** [comment]
Job PR-705/4 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-705/4/index.html

**Comment by mli** [comment]
Job PR-705/5 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-705/5/index.html

**Comment by mli** [comment]
Job PR-705/6 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-705/6/index.html

**Comment by mli** [comment]
Job PR-705/7 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-705/7/index.html

File: scripts/bert/export/export.py [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+                         '"wiki_multilingual_uncased", "wiki_multilingual_cased", '
+                         '"wiki_cn_cased"')
+
+parser.add_argument('--output_dir',
+                    type=str,
+                    default='./output_dir',
+                    help='The directory where the exported model symbol will be created. '
+                         'The default is ./output_dir')
+
+parser.add_argument('--seq_length',
```
Comment (pengxin99) [inline scripts/bert/export/export.py]: A small question, don`t we need the "input_size" parameter here? like [static_finetune_squad.py ](https://github.com/dmlc/gluon-nlp/blob/c6bf1c5b078a37fb6e9805a9c5ce86d4595ad5d2/scripts/bert/staticbert/static_finetune_squad.py#L197)

File: scripts/bert/export/export.py [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+    inputs, token_types, valid_length = batch
+    net(inputs, token_types, valid_length)
+    net.export(prefix, epoch=0)
+    assert os.path.isfile(prefix + '-symbol.json')
+    assert os.path.isfile(prefix + '-0000.params')
+
+def infer(batch, prefix):
+    """Evaluate the model on a mini-batch."""
+    log.info('Start inference ... ')
+    tic = time.time()
```
Comment (pengxin99) [inline scripts/bert/export/export.py]: Maybe the model import time not be included in inference time,  should we count the tic after this`mx.gluon.nn.SymbolBlock.imports()`  imports code?

File: scripts/bert/export/hybrid_bert.py [hunk 3]
```diff
@@ -530,7 +531,7 @@ def _apply_pooling(self, sequence):
 
         This is used for pre-training or fine-tuning a BERT model.
         """
-        outputs = sequence[:, 0, :]
+        outputs = sequence.slice(begin=(None, 0, None), end=(None, 1, None)).squeeze(axis=1)
         return self.pooler(outputs)
 
     def _decode(self, sequence, masked_positions):
```
Comment (pengxin99) [inline scripts/bert/export/hybrid_bert.py]: Because we won`t use the decoder at bert, so can we delete the decoder part or set 
> use_decoder : bool, default True
        Whether to include the decoder for masked language model prediction.

 default false? i think it will make our bert model clearer.

**Comment by TaoLv** [comment]
Thank you for the contribution. I will have a try!

File: scripts/bert/export/export.py (hunk 1 above)
Comment (eric-haibin-lin) [reply scripts/bert/export/export.py]: I don't think that's necessary. The embedding size should be fixed for either bert base or bert large.

File: scripts/bert/export/export.py (hunk 2 above)
Comment (eric-haibin-lin) [reply scripts/bert/export/export.py]: Thanks for pointing that out. I'll update it

File: scripts/bert/export/hybrid_bert.py (hunk 3 above)
Comment (eric-haibin-lin) [reply scripts/bert/export/hybrid_bert.py]: Good observation. I try to be consistent with the get_bert_model in the gluonnlp API: https://github.com/dmlc/gluon-nlp/blob/master/src/gluonnlp/model/bert.py#L630 

It might be true that most people use BERT for finetuning and don't need the decoder. But it's not likely that we'll break existing API before gluonnlp 1.0.

**Comment by mli** [comment]
Job PR-705/8 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-705/8/index.html

**Comment by mli** [comment]
Job PR-705/9 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-705/9/index.html

**Comment by xinyu-intel** [comment]
@eric-haibin-lin How to run classification model inference?

**Comment by gigasquid** [comment]
Thanks @eric-haibin-lin ! I'll help test the export base script.

File: scripts/bert/index.rst [hunk 4]
```diff
@@ -319,153 +319,14 @@ Command line interface
...
-| command               | [8]                                                                                                                        | [9]                                                                                                                         | [10]                                                                                                                        |
-+-----------------------+----------------------------------------------------------------------------------------------------------------------------+-----------------------------------------------------------------------------------------------------------------------------+-----------------------------------------------------------------------------------------------------------------------------+
+    $ python export/export.py --task classification --model_parameters /path/to/saved/ckpt.params --output_dir /path/to/output/dir/ --seq_length 256
```
Comment (xinyu-intel) [inline scripts/bert/index.rst]: why 256 here? i find 128 is a default value in finetune_classification.py.

Comment (eric-haibin-lin) [reply scripts/bert/index.rst]: I just picked an arbitrary number. I can update it

**Comment by mli** [comment]
Job PR-705/10 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-705/10/index.html

**Comment by harplife** [comment]
Hi, is there any guide on how to use the export script? I would like to export a fine-tuned Bert model.

The code I'm using to train the model is [this](https://gist.github.com/haven-jeon/3d7c538398e93dab2ed4899159a5d943).
